storage/watchable_store_test.go

```diff
@@ -222,3 +234,95 @@ func TestUnsafeAddWatching(t *testing.T) {
 		}
 	}
 }
+
+func TestMapWatchingToEvents(t *testing.T) {
```

Comment:
you do not need that 100 keys. Do you? Normally you can manually populate the input and output without programmatic do it. Basically we want the test cases to cover all the cases.